osu.Framework/Bindables/Bindable.cs

```
@@ -262,7 +272,15 @@ public virtual void Parse(object input)
                     Value = bindable.Value;
                     break;
 
+                case string str when string.IsNullOrEmpty(str):
+                    // This is slightly different from the `null` case.
+                    // Consider a non-nullable default-initialised struct for which `ToString()` outputs an empty string.
+                    Value = default;
+                    break;
```

I find this case... particularly peculiar. It is operating on a lot of assumptions (e.g. that for every single non-nullable bindable there can ever be, an empty string is allowed to coerce to that bindable's inner type's `default`). Is this needed for anything?

I find the test cases that exercise this path kinda bizarre. `Bindable<int>` parsing empty strings into 0 feels just straight up weird. Built-in `int.Parse()` vomits on empty strings.

---

To compound the above, I wrote the following test:

```diff
diff --git a/osu.Framework.Tests/Bindables/BindableTest.cs b/osu.Framework.Tests/Bindables/BindableTest.cs
index 9fdb63118..8d52a5d1b 100644
--- a/osu.Framework.Tests/Bindables/BindableTest.cs
+++ b/osu.Framework.Tests/Bindables/BindableTest.cs
@@ -196,6 +196,24 @@ public void TestParseEmptyStringIntoNullableReferenceTypeWithNRT()
             bindable.Parse(string.Empty);
             Assert.That(bindable.Value, Is.Null);
         }
+
+        // Bindable<string>.Parse(string.Empty)
+        [Test]
+        public void TestParseEmptyStringIntoNonNullableString()
+        {
+            Bindable<string> bindable = new Bindable<string>();
+            bindable.Parse(string.Empty);
+            Assert.That(bindable.Value, Is.Empty);
+        }
+
+        // Bindable<string>.Parse(string.Empty)
+        [Test]
+        public void TestParseEmptyStringIntoNullableString()
+        {
+            Bindable<string?> bindable = new Bindable<string?>();
+            bindable.Parse(string.Empty);
+            Assert.That(bindable.Value, Is.Empty);
+        }
 #nullable disable
 
         private static IEnumerable<object[]> getParsingConversionTests()
```

While they _pass_, they pass by what appears to either be sheer luck, or very careful placement. In the case of `Bindable<string>`, the `case T t` label is picked rather than the `case string str when string.IsNullOrEmpty(str)` label.

Maybe this was intentional, and maybe we want to keep this behaviour, but even if both are true, I'd say that maybe we want the above test cases to be in code too, just in case.

---

At a high level, I'd prefer matching the behaviour of .NET parse methods (throwing rather than graceful `default`). But need more context on whether this was done for a specific reason as per above.

---

That's fair. So in your view all `string.Empty` cases should throw the same exception as `int.Parse()`, but they're fine to keep for nullables? In my particular case, as my OP mentions, I only need the `string.Empty` -> `null` coercion but maybe there's a better way of doing it that I haven't discovered yet.

---

For non-string nullable types, I can agree with parsing `string.Empty` to `null` for sure.

---

I've adjusted the implementation a bit, see if you agree with the new results.